internal/services/loganalytics/log_analytics_workspace_resource.go

```diff
@@ -221,7 +220,7 @@ func resourceLogAnalyticsWorkspaceCreateUpdate(d *pluginsdk.ResourceData, meta i
 	t := d.Get("tags").(map[string]interface{})
 
 	if isLACluster {
-		sku.Name = "lacluster"
+		sku.Name = workspaces.WorkspaceSkuNameEnumLACluster
```

Comment:
I believe this maybe a breaking change depending on how the new enum cases the value. In the previous version we used `lacluster`. So, if there is an existing resource with the sku `lacluster` and the new value in the enum is `LACluster` that my cause an issue here?

Reply:
We'll need to do a state migration and fix it on read perhapes?

Reply:
@ziyeqf - Are you able to check if this is an issue (i.e. build one with a released version, then manage it with your updated code) and add the state migration if needed?

Reply:
My test steps:
1. deploy with latest released version.
2. run `terraform plan` and then a `ForceNew` diff shows.
```
      ~ sku                                = "LACluster" -> "PerGB2018" # forces replacement
```
3. plan with updated code, a diff without `ForceNew` shows.
```
      ~ sku                        = "LACluster" -> "PerGB2018"
```
5. apply with updated code, it worked as expected.

My explain is:
After changed to azure-go-sdk, it has begun to set `LACluster` to state file, which was `""` before, then the workaround goes invalid. 

For the casing change here, it's only between provider and service, we used to use `lacluster` in payload, and `LACluster` now. Per the test result I think it should be ok?

Code ref:
https://github.com/hashicorp/terraform-provider-azurerm/blob/4ce0783b21fd4c07c4160aae37e910a5a8708870/internal/services/loganalytics/log_analytics_workspace_resource.go#L382-L394

Constant ref:
https://github.com/hashicorp/go-azure-sdk/blob/10bd4eb4bbb6822a564b2d0fcdf348e8ed3a8e8c/resource-manager/operationalinsights/2020-08-01/workspaces/constants.go#L177-L186

Reply:
Hi @ziyeqf - I think you should test with a `lacluster` value and check the effect of the change to `LACluster`, since this Enum change will likely result in a diff, which is what I think @WodansSon was pointing out? This may need a state migration to update the value to prevent this.
Thanks!

Reply:
@jackofallops I don't think there will be a config with `sku` set to `lacluster`, it's now allowed in `validateFunc` and will throw error in plan stage. And for state file, it will be set to `LACluster` in refresh stage.

Reply:
> @jackofallops I don't think there will be a config with `sku` set to `lacluster`, it's not allowed in `validateFunc` and will throw error in plan stage. And for state file, it will be set to `LACluster` in refresh stage.

sorry for typo, it's "not" allowed.

Reply:
previous versions allowed lacluster - we'll need to account for that as we don't know what version someone will be upgrading from so we can't make that asausmption

Reply:
So.. we need to allow `lacluster` in `validateFunc` again?

Reply:
Apologies for the back and forth @ziyeqf. Your implementation here is correct. We are only setting `lacluster` during create and update and the api doesn't care about casing. During the read, we are setting `sku` to `LACluster` so the end user shouldn't encounter any state changes from this PR